src/client.rs

```diff
@@ -19,6 +19,18 @@ use byteorder::{BigEndian, ByteOrder};
 use retry::{delay::Fixed, retry};
 use std::io::{self, BufReader, BufWriter, Read};
 use std::net::TcpStream;
+use std::time::Duration;
+use std::{env, net::ToSocketAddrs};
+
+const DEFAULT_CONNECTION_TIMEOUT: u64 = 20;
```

Comment:
This could be much lower, but leaving it this high aligns it more closely with the current behavior. I'm running it with `2` locally.